index.js

```diff
@@ -348,42 +295,81 @@ function asThrift(opts, request, onResponse) {
     try {
         sender.send(request, opts.endpoint, opts.head,
             opts.body, onResponse);
-    } catch(e) {
-        if (e.message ===
-            fmt('type %s_args not found', opts.endpoint)) {
-            var emsg = fmt('%s endpoint does not exist', opts.endpoint);
-            onResponse(new Error(emsg));
+    } catch (err) {
+        // TODO untangle this mess
+        if (err.message === fmt('type %s_args not found', opts.endpoint)) {
+            delegate.error(fmt('%s endpoint does not exist', opts.endpoint));
+            done();
+            return delegate.exit();
         } else {
-            var msg = e.message || '';
-            reportError(opts.logger,
-                'Error response received for the as-thrift request. '
-                + msg, e);
+            delegate.error('Error response received for the as-thrift request.');
+            delegate.error(err);
+            done();
+            return delegate.exit();
         }
     }
-}
 
-function asRaw(opts, request, onResponse) {
-    request.headers.as = 'raw';
+    function onResponse(err, res, arg2, arg3) {
+        done();
+        self.onResponse(err, res, arg2, arg3, opts, delegate);
+    }
+};
+
+TCurl.prototype.asRaw = function asRaw(opts, request, delegate, done) {
+    var self = this;
+    request.headers.as = opts.as || 'raw';
```

Comment:
Yes, for now. I’ll fix this.

The motivation for this change is that we would like to eventually be able to pipe an arg or args out in their “raw” form, e.g., to pipe to xxd to inspect Thrift binary. I will leave this for a future exploration.